app/invocation/invocation_action_card.tsx

```diff
@@ -127,65 +128,126 @@ export default class ActionCardComponent extends React.Component<Props, State> {
+                      <div className="action-property-title">Environment Variables:</div>
+                      <div className="action-list">
+                        {this.state.command.environmentVariables.map((variable) => (
+                          <div>
+                            <span className="prop-name">{variable.name}</span>
+                            <span className="prop-value">={variable.value}</span>
+                          </div>
+                        ))}
+                      </div>
                     </div>
                   </div>
-                  <div className="action-section">
-                    <div className="action-property-title">Output Directories:</div>
-                    {this.displayList(this.state.command.outputDirectories)}
-                  </div>
-                  <div className="action-section">
-                    <div className="action-property-title">Output Files:</div>
-                    {this.displayList(this.state.command.outputFiles)}
-                  </div>
-                  <div></div>
-                </div>
-              )}
-              {this.state.actionResult && (
-                <div>
-                  <pre>
-                    <code>{JSON.stringify(this.state.actionResult, null, 2)}</code>
-                  </pre>
+                )}
+              </div>
+              <div className="action-line">
+                <div className="action-title">Result details</div>
+                {this.state.actionResult && (
                   <div>
-                    {this.state.actionResult.outputDirectories.map((dir) => (
-                      <div>
-                        <div>{dir.path}</div>
-                        <div>{dir.treeDigest.hash}</div>
-                        <div>{dir.treeDigest.hash}</div>
-                      </div>
-                    ))}
+                    <div className="action-section">
+                      <div className="action-property-title">Exit Code: </div>
+                      <div>{this.state.actionResult.exitCode}</div>
+                    </div>
+                    <div className="action-section">
+                      <div className="action-property-title">Execution Metadata:</div>
+                      {this.state.actionResult.executionMetadata ? (
+                        <div className="action-list">
+                          <div className="metadata-title">Worker</div>
+                          <div className="metadata-detail">{this.state.actionResult.executionMetadata.worker} </div>
+                          <div className="metadata-title">Executor ID</div>
+                          <div className="metadata-detail">{this.state.actionResult.executionMetadata.executorId}</div>
+                          <div className="metadata-title">Timeline</div>
+                          <div className="metadata-detail">
+                            Queued @ {this.state.actionResult.executionMetadata.queuedTimestamp.nanos}
```

Comment:
you should just be able to add another formatter here that supports nanos. totally cool if you want to do that in a separate PR

https://github.com/buildbuddy-io/buildbuddy/blob/master/app/format/format.tsx#L14